### PR TITLE
 Added handling of fb:part logical forms and linking

### PR DESCRIPTION
**Pull request by matt-gardner**
This gets most of the missing `fb:part.[whatever]` logical forms.  There are still some odd errors where we encounter unmapped names when parsing logical forms, but there aren't that many of them now.  I added parts as entities in the `KnowledgeGraph`, so our linking code will handle them, but they have no neighbors in the graph, so the neighbor code _should_ just ignore them.

This PR depends on #936, which in turn depends on #915.

### CODE REVIEW

**Comment by matt-gardner** [comment]
Ok, this is now ready for review.

File: allennlp/data/semparse/knowledge_graphs/table_question_knowledge_graph.py [hunk 1]
```diff
@@ -250,3 +285,40 @@ def _get_numbers_from_tokens(tokens: List[Token]) -> List[Tuple[str, str]]:
...
+    def _get_cell_parts(cls, cell_text: str) -> List[Tuple[str, str]]:
+        """
+        Splits a cell into parts and returns the parts of the cell.  We return a list of
+        ``(entity_name, entity_text)``, where ``entity_name`` is ``fb:part.[something]``, and
+        ``entity_text`` is the text of the cell corresponding to that part.  For many cells, there
+        is only one "part", and we return a list of length one.
+
+        Note that you shouldn't call this on every cell in the table; SEMPRE decides to make these
+        splits only when at least one of the cells in a column looks "splitable".  Only if you're
+        splitting a column should you use this function.
```
Comment (pdasigi) [inline allennlp/data/semparse/knowledge_graphs/table_question_knowledge_graph.py]: splitting a column -> splitting cells in a column

File: allennlp/data/semparse/knowledge_graphs/table_question_knowledge_graph.py [hunk 2]
```diff
@@ -250,3 +285,40 @@ def _get_numbers_from_tokens(tokens: List[Token]) -> List[Tuple[str, str]]:
...
+        """
+        parts = []
+        for part_text in cls.cell_part_regex.split(cell_text):
+            part_text = part_text.strip()
+            part_entity = f'fb:part.{cls._normalize_string(part_text)}'
+            parts.append((part_entity, part_text))
+        return parts
+
+    @classmethod
+    def _should_split_column(cls, column_cells: List[str]) -> bool:
```
Comment (pdasigi) [inline allennlp/data/semparse/knowledge_graphs/table_question_knowledge_graph.py]: May be call this method `_should_split_column_cells`?

File: allennlp/data/semparse/knowledge_graphs/table_question_knowledge_graph.py (hunk 1 above)
Comment (matt-gardner) [reply allennlp/data/semparse/knowledge_graphs/table_question_knowledge_graph.py]: Thanks.

File: allennlp/data/semparse/knowledge_graphs/table_question_knowledge_graph.py (hunk 2 above)
Comment (matt-gardner) [reply allennlp/data/semparse/knowledge_graphs/table_question_knowledge_graph.py]: Good point, done.

**Pull request merged**